src/Reflex/Process.hs

```diff
@@ -127,16 +143,21 @@ createProcess
   => CreateProcess
   -> ProcessConfig t ByteString
   -> m (Process t ByteString ByteString)
-createProcess = createRedirectedProcess input output output
-  where
+createProcess p processConfig = do
+  channel <- liftIO newTChanIO
+
+  let
     input h = do
       H.hSetBuffering h H.NoBuffering
-      let go b = do
-            open <- H.hIsOpen h
-            when open $ do
-              writable <- H.hIsWritable h
-              when writable $ Char8.hPutStrLn h b
-      return go
+      void $ liftIO $ forkIO $ fix $ \loop -> do
+        newMessage <- atomically $ readTChan channel
+        open <- H.hIsOpen h
+        when open $ do
+          writable <- H.hIsWritable h
+          when writable $ do
+            Char8.hPutStrLn h newMessage
```

Comment:
I don't think we want to use `Char8` here even though I'm not sure there's actually any error. But we definitely don't want to be shedding any doubt about whether this is suitable for arbitrary bytes.

Reply:
Oh my I just realized this is using `hPutStrLn` which is a bug anyway. That's going to insert newlines in random places. We should leave the line-breaking to the input stream itself.